### PR TITLE
feat: add adapter for hertz

### PR DESCRIPTION
**Pull request by Skyenought**
### Describe what this PR does / why we need it
https://github.com/cloudwego/hertz/issues/138

### Does this pull request fix one issue?
https://github.com/cloudwego/hertz/issues/138

### Describe how you did it
The implementation is based on hertz middlewares for http server and client.

They have been tested in `server_test.go` and `client_test.go`

### Describe how to verify it
The implementation can be verified via `go test`.

### Special notes for reviews

### CODE REVIEW

File: pkg/adapters/hertz/.gitignore [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# JetBrains template
```
Comment (binbin0325) [inline pkg/adapters/hertz/.gitignore:1]: gitignore is not required, it is already available in the main project

Comment (Skyenought) [reply pkg/adapters/hertz/.gitignore:1]: GET

**Review by binbin0325 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Nice work. Thanks for contributing!

**Pull request merged**